elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/core/JoinPath.java

```diff
@@ -61,4 +63,43 @@ public JoinPath extend(String extensionPath, EntityDictionary dictionary) {
         toExtend.addAll(extension.getPathElements());
         return new JoinPath(toExtend);
     }
+
+    /**
+     * Resolve a dot separated path into list of path elements.
+     *
+     * @param entityClass root class e.g. "foo"
+     * @param dictionary dictionary
+     * @param dotSeparatedPath path e.g. "bar.baz"
+     * @return list of path elements e.g. ["foo.bar", "bar.baz"]
+     */
+    private List<PathElement> resolvePathElements(Class<?> entityClass,
+                                                  EntityDictionary dictionary,
+                                                  String dotSeparatedPath) {
+        List<PathElement> elements = new ArrayList<>();
+        String[] fieldNames = dotSeparatedPath.split("\\.");
+
+        Class<?> currentClass = entityClass;
+        for (String fieldName : fieldNames) {
+            if (needNavigation(currentClass, fieldName, dictionary)) {
+                Class<?> joinClass = dictionary.getParameterizedType(currentClass, fieldName);
+                elements.add(new PathElement(currentClass, joinClass, fieldName));
+                currentClass = joinClass;
+            } else {
+                elements.add(resolvePathAttribute(currentClass, dictionary, fieldName));
+            }
+        }
+
+        return ImmutableList.copyOf(elements);
+    }
+
+    private PathElement resolvePathAttribute(Class<?> entityClass,
+                                             EntityDictionary dictionary,
+                                             String fieldName) {
+        Class<?> attributeClass = null;
```

Comment:
Since we don't know the field type, we could make it Object.class.

Reply:
updated.